inc/saihostintf.h

```diff
@@ -692,6 +587,179 @@ typedef sai_status_t(*sai_get_hostif_attribute_fn)(
+*/
+typedef enum _sai_host_interface_table_entry_channel_type_t
+{
+    /** receive packets via callback */
+    SAI_HOST_INTERFACE_TABLE_ENTRY_CHANNEL_TYPE_CB,
+
+    /** receive packets via file descriptor */
+    SAI_HOST_INTERFACE_TABLE_ENTRY_CHANNEL_TYPE_FD,
+
+    /** receive packets via Linux netdev type port */
+    SAI_HOST_INTERFACE_TABLE_ENTRY_CHANNEL_TYPE_NETDEV_PHYSICAL_PORT,
+
+    /** receive packets via Linux netdev logical port (LAG or port) */
+    SAI_HOST_INTERFACE_TABLE_ENTRY_CHANNEL_TYPE_NETDEV_LOGICAL_PORT,
+
+    /** receive packets via Linux netdev L3 interface */
+    SAI_HOST_INTERFACE_TABLE_ENTRY_CHANNEL_TYPE_NETDEV_L3
+
+} sai_host_interface_table_entry_channel_type_t;
+
+/**
+* @brief Host interface table entry attribute IDs
+*/
+typedef enum _sai_hostif_table_entry_attr_t
+{
+    /**
+    * @brief Start of attributes
+    */
+    SAI_HOSTIF_TABLE_ENTRY_ATTR_START,
+
+    /**
+    * @brief Host interface table entry type
+    *
+    * @type sai_host_interface_table_entry_type_t
+    * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+    */
+    SAI_HOSTIF_TABLE_ENTRY_ATTR_TYPE = SAI_HOSTIF_ATTR_START,
+
+    /**
+    * @brief Host interface table entry match field object-id
+    *
+    * Valid only when #SAI_HOSTIF_TABLE_ENTRY_ATTR_TYPE == #SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_PORT
+    * || #SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_LAG || #SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_VLAN
+    * should be port object when type is SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_PORT
+    * should be lag object when type is SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_LAG
+    * should be VLAN ID object when type is SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_VLAN
+    *
+    * @type sai_object_id_t
+    * @objects SAI_OBJECT_TYPE_PORT, SAI_OBJECT_TYPE_LAG, SAI_OBJECT_TYPE_ROUTER_INTERFACE
+    * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+    * @condition SAI_HOSTIF_TABLE_ENTRY_ATTR_TYPE == SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_PORT or SAI_HOSTIF_TABLE_ENTRY_ATTR_TYPE == SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_VLAN or SAI_HOSTIF_TABLE_ENTRY_ATTR_TYPE == SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_LAG
+    */
+    SAI_HOSTIF_TABLE_ENTRY_ATTR_OBJ_ID,
+
+    /**
+    * @brief Host interface table entry match field trap-id
+    *
+    * Valid only when #SAI_HOSTIF_TABLE_ENTRY_ATTR_TYPE == #SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_PORT || 
+    * #SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_LAG || #SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_VLAN ||
+    * #SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_TRAP_ID
+    * @type sai_object_id_t
+    * @objects SAI_OBJECT_TYPE_HOSTIF_TRAP
+    * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+    * @condition SAI_HOSTIF_TABLE_ENTRY_ATTR_TYPE == SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_PORT or SAI_HOSTIF_TABLE_ENTRY_ATTR_TYPE == SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_VLAN or SAI_HOSTIF_TABLE_ENTRY_ATTR_TYPE == SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_LAG or SAI_HOSTIF_TABLE_ENTRY_ATTR_TYPE == SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_TRAP_ID
+    */
+    SAI_HOSTIF_TABLE_ENTRY_ATTR_TRAP_ID,
```

Comment:
trap id has a exclude port list. Hostif table entry has a port (SAI_HOSTIF_TABLE_ENTRY_ATTR_OBJ_ID), how do they interact? If I have a port here that is excluded in the trap what is expected.